lua/nordic/groups/init.lua

```diff
@@ -1,11 +1,11 @@
-local merge = require('nordic.utils').merge
+local merge_inplace = require('nordic.utils').merge_inplace
 
 local M = {}
 
 function M.get_groups()
     local native = require('nordic.groups.native').get_groups()
     local integrations = require('nordic.groups.integrations').get_groups()
-    local groups = merge(native, integrations)
+    local groups = merge_inplace(native, integrations)
```

Comment:
Pretty sure this should not be a merge inplace?  We do not want to modify `native` here.

Reply:
It works because `get_groups` creates a new table each time; however, I agree it's not very clear. `merge` and `merge_inplace` do almost the same thing, so it feels weird to have both.


I could revert that commit or I could make it more clear with something like this:

```lua
local native = require('nordic.groups.native').get_groups()
local integrations = require('nordic.groups.integrations').get_groups()

local groups = {}
merge_inplace(groups, integrations)
merge_inplace(groups, integrations)
```

Which ever you think is better.

Reply:
Merge and merge inplace are difference enough that we would want both, it is very common to have both.

So I think we should:
- Keep both.
- If you want to keep using `merge_inplace` in this scenario, please change it the way you suggested.  Just a bit more clear what is going on.

Reply:
done in: 3da1cff